docs/howtos/customizations/metrics/_train_your_own_metric.md

```diff
@@ -0,0 +1,162 @@
+
+
+## Import required modules
+
+
+```python
+import os
+from datasets import load_dataset
+from ragas import evaluate, EvaluationDataset
+from ragas.metrics import AspectCritic
+
+```
+
+### Setup the models used for evaluation and training
+You may choose any LLM model for training and evaluation. Here's [how to do it](../customize_models.md)
+
+
+```python
+from ragas.llms import LangchainLLMWrapper
+from ragas.embeddings import LangchainEmbeddingsWrapper
+from langchain_openai import ChatOpenAI
+from langchain_openai import OpenAIEmbeddings
+llm = LangchainLLMWrapper(ChatOpenAI(model="gpt-4o-mini"))
+embeddings = LangchainEmbeddingsWrapper(OpenAIEmbeddings())
+```
+
+## Evaluation
+
+### Load sample evaluation dataset
+Here, we are loading the sample dataset for evaluation. You can replace it with your own dataset. 
+
+
+```python
+dataset = load_dataset("explodinggradients/ELI5",split="test")
+eval_dataset = EvaluationDataset.from_hf_dataset(dataset)
+```
+
+### Setup the Metric
+You may use any LLM based metric. For simplicity, I am using aspect critic metric and setting it up so that it can compare the response with the reference.
+
+
+```python
+critic = AspectCritic(name="answer_correctness",definition="Given the user_input, reference and response. Is the response correct compared with the reference",llm=llm)
+
+```
+
+### Evaluate
+
+
+```python
+results = evaluate(eval_dataset,metrics=[critic])
+
+```
+
+    Evaluating: 100%|██████████████████████████████████████████████████████████| 56/56 [00:01<00:00,  3.49it/s]
+
+
+
+## Review and Annotate
+
+Now you have the evaluation results. Now it's time to review the evaluations and give feedback to the metric. This feedback will be used to train the metric. For this you can use [app.ragas](https://app.ragas.io) or any other annotation tool like prodigy, label studio etc.
+
+If you're using app.ragas,
+
+- Go to [app.ragas](https://app.ragas.io) and login.
+- Then go to [App tokens](https://app.ragas.io/dashboard/settings/app-tokens) and create a new app token.
+- Set the token in the environment variable
+
+```python
+os.environ['RAGAS_APP_TOKEN'] = 'your_app_token'
+```
+
+Once that's done, you can upload the evaluation results to app.ragas using the following code.
+
+```python
+results.upload()
+```
+
+Evaluation results uploaded! View at https://app.ragas.io/dashboard/alignment/evaluation/a6baf6ff-027f-4097-89e3-e11c70b8cf61
+'https://app.ragas.io/dashboard/alignment/evaluation/a6baf6ff-027f-4097-89e3-e11c70b8cf61'
+
+
+
+### Review and annotate some results
+You may now view and annotate the evaluation results in app.ragas. These annotations will be used to train the metric. Please make sure to annotate at least 15-20 examples for good results.
+
+<figure markdown="span">
+  ![Annotating](../../../_static/imgs/annotation.png){width="600"}
+  <figcaption>Annotating results</figcaption>
+</figure>
+
+
```

Comment:
should we provide an annotated file here? would helpout users who might not annotate the samples?

Reply:
pros and cons. If we do that half of the users may not sign up for app.ragas

Reply:
Anyway, will add one

Reply:
@jjmachan I have added .json file to git here. Is that a problem?